cloudinit/config/cc_ansible.py

```diff
@@ -136,13 +136,32 @@ def install(self, pkg_name: str):
                 import pip  # noqa: F401
             except ImportError:
                 self.distro.install_packages(self.distro.pip_package_name)
+
+            pip_version_output = self.do_as(
```

Comment:
Instead of checking pip version to see if we can provide this output, shouldn't we instead be checking if the EXTERNALLY-MANAGED marker file exists in the first place? This could be our indicator that the system is implementing the PEP 668, and therefore probably also supports --break-system-packages param. This would also have the benefit on stable releases of no change in behavior if the distro itself doesn't package the marker file.

[Pip does this check like so](https://github.com/pypa/pip/blob/e6deb9b87c18cdd27a9ba27cb7e0670ffb81d45e/src/pip/_internal/utils/misc.py#L594C1-L595C1)  The additional benefit of doing a file presence check is that it avoids a more costly shell out to the cmdline.

Reply:
> Instead of checking pip version to see if we can provide this output, shouldn't we instead be checking if the EXTERNALLY-MANAGED marker file exists in the first place?

@blackboxsw This file is not a requirement, but rather a [recommendation for distros to include](https://peps.python.org/pep-0668/#recommendations-for-distros). This comes with distro risk. Your recommendation is cleaner, however, so I'm slightly leaning towards doing that anyway. Thoughts?